File: app/src/main/java/com/example/bmi/ChartsScreen.java
package com.example.bmi;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.widget.Button;

public class ChartsScreen extends AppCompatActivity {

    private Button buttonBar;
    private Button buttonPie;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.charts_screen);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        buttonBar = (Button) findViewById(R.id.button2);
        buttonBar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openBarChart();
            }
        });

        buttonPie = (Button) findViewById(R.id.button3);
        buttonPie.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openPieChart();
            }
        });
    }

    public void openBarChart() {
        Intent intent = new Intent(this, BarChartScreen.class);
        startActivity(intent);
    }

    public void openPieChart() {
        Intent intent = new Intent(this, PieChartScreen.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/bmi_kotlin/ChartsScreen.kt
package com.example.bmi_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class ChartsScreen : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.charts_screen)
        setSupportActionBar(findViewById(R.id.toolbar))
        val buttonBar = findViewById<Button>(R.id.button2)
        buttonBar!!.setOnClickListener { openBarChart() }
        val buttonPie = findViewById<Button>(R.id.button3)
        buttonPie!!.setOnClickListener { openPieChart() }
    }

    fun openBarChart() {
        val intent = Intent(this, BarChartScreen::class.java)
        startActivity(intent)
    }

    fun openPieChart() {
        val intent = Intent(this, PieChartScreen::class.java)
        startActivity(intent)
    }
}